flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -39,10 +39,53 @@
 			key("taskmanager.jvm-exit-on-oom")
 			.defaultValue(false);
 
+	/** Size of memory buffers used by the network stack and the memory manager (in bytes). */
+	public static final ConfigOption<Integer> MEMORY_SEGMENT_SIZE =
+			key("taskmanager.memory.segment-size")
+			.defaultValue(32768);
+
+	/**
+	 * Amount of memory to be allocated by the task manager's memory manager (in megabytes). If not
+	 * set, a relative fraction will be allocated, as defined by {@link #MANAGED_MEMORY_FRACTION}.
+	 */
+	public static final ConfigOption<Long> MANAGED_MEMORY_SIZE =
+			key("taskmanager.memory.size")
+			.defaultValue(-1L);
```

Comment:
This seems like an odd default value since it isn't actually a valid value for it. All usages where the value is -1 have a special case for it; `.noDefaultValue()` seems more appropriate.